src/Expander/Monad.hs

```diff
@@ -365,11 +367,19 @@ modifyState f = do
   !st <- view expanderState <$> expanderContext
   liftIO (modifyIORef' st f)
 
+#ifndef KDEBUG
+freshScope :: Expand Scope
+freshScope = do
+  n <- view expanderNextScopeNum <$> getState
+  modifyState $ over expanderNextScopeNum (+ 1)
+  return (Scope n)
```

Comment:
I like the idea, but this would require decoupling `Expand` from the `Monad` module because the smart constructor needs to know about the type `Text -> Expand Scope`, so if I make a module to encapsulate the smart constructor then it must import `Expander.Monad` in order to gain access to the `Expand` type, but `Expander.Monad` is exactly the module I want to import the smart constructor to. Thus we get a module cycle.